### DISCUSSION TITLE
multi-step form with `useFieldArray`. `dirtyFields` not working as expected

### DISCUSSION THREAD

**Discussion by alicanerdurmaz** [react-hook-form/react-hook-form#10290] [Q&A] [answered]
Hello everyone,
I am trying to create an edit multi-step form with `useFieldArray`. 

What I am trying to achieve inside the `useEffect` is to set the data that comes from the API in each step and allow the user to edit the data.

When I use `useFieldArray` and append a new field to the array, the dirty state of the fields in the other steps becomes `false`, which prevents me from achieving the desired behavior. If I don't use `useFieldArray`, there is no problem.

What am I doing wrong?

example with `useFieldArray`: https://codesandbox.io/s/beautiful-mahavira-pvy4e3
example without `useFieldArray`: https://codesandbox.io/s/zen-brook-z6cpzt

**Comment by leapful** [comment, marked as answer]
@alicanerdurmaz 

`reset` method will try to set default value for the target field name and also trigger dirty check on next re-rendering. In your case, dirty value on `title` or `content` will become default value on next re-rendering and therefore it will be no more dirty to the form.

I would suggest to use `setValue` at `useLayoutEffect` to update value before field is being registered to achieve same result.

Working demo: https://codesandbox.io/s/bold-sky-pbmemq?file=/src/App.tsx
